components/src/core/DropdownToolbar/DropdownToolbar.tsx

```diff
@@ -0,0 +1,181 @@
+    const theme = useTheme();
+    const defaultClasses = useStyles(theme);
+    const getNavigationIcon = useCallback(() => {
+        if (navigationIcon) {
+            return (
+                <div className={clsx(defaultClasses.navigationIconWrapper, classes?.navigationIconWrapper)}>
+                    {navigationIcon}
+                </div>
+            );
+        }
+    }, [navigationIcon]);
+
+    return (
+        <>
+            <Toolbar data-testid="dropdown-toolbar" className={clsx(defaultClasses.root, classes?.root)} style={style}>
+                {getNavigationIcon()}
+                <ListItemText
+                    data-testid="dropdown-title"
+                    id={'dropdown-toolbar-text'}
+                    primary={
+                        <Typography variant="h6" className={clsx(defaultClasses.title, classes?.title)}>
+                            {title}
+                        </Typography>
+                    }
+                    secondary={
+                        <Typography
+                            id={'dropdown-subtitle'}
+                            data-testid={'dropdown-subtitle'}
+                            ref={anchor}
+                            aria-haspopup="true"
+                            className={clsx(defaultClasses.subtitle, classes?.subtitle)}
+                            component={'div'}
+                            onClick={(): void => {
+                                setAnchorEl(anchor.current);
+                            }}
+                        >
+                            {subtitleLabel || ''}
+                            {menuItems && menuItems.length > 0 && (
+                                <ArrowDropDown className={clsx(defaultClasses.arrowDropdown, classes?.arrowDropdown)} />
+                            )}
+                            {customMenu && (
+                                <ArrowDropDown className={clsx(defaultClasses.arrowDropdown, classes?.arrowDropdown)} />
+                            )}
+                        </Typography>
+                    }
+                />
+                {props.children}
+            </Toolbar>
+
+            {menuItems && Boolean(anchorEl) && (
+                <Menu
+                    id="dropdown-toolbar-menu"
+                    data-testid="dropdown-toolbar-menu"
+                    elevation={0}
+                    getContentAnchorEl={null}
+                    anchorOrigin={{ vertical: 'bottom', horizontal: 'center' }}
+                    transformOrigin={{ vertical: 'top', horizontal: 'center' }}
+                    anchorEl={anchorEl}
+                    open={Boolean(anchorEl)}
+                    onClose={(): void => setAnchorEl(null)}
+                    MenuListProps={{ style: { padding: 0 } }}
+                    PaperProps={{
+                        style: {
+                            maxHeight: MENU_ITEM_HEIGHT * 4.5,
+                            width: 200,
```

Comment:
Might not want to set a fixed width on this since we aren't in control of what the labels are (WAS knew what they would be in advance)